example/main.go

```diff
@@ -2,152 +2,262 @@ package main
+		awsAlgorithm, credential, signedHeaders, signature)
+
+	req.Header.Set("Authorization", authHeader)
+
+	// Debug output
+	fmt.Println("\n🔐 Request Headers:")
+	for k, v := range req.Header {
+		fmt.Printf("%s: %s\n", k, v)
 	}
 
-	// Make request with transport client
-	start := time.Now()
+	// Create HTTP client and make request
+	client := &http.Client{}
 	resp, err := client.Do(req)
 	if err != nil {
-		log.Fatalf("Request failed: %v", err)
+		return "", fmt.Errorf("request failed: %v", err)
 	}
 	defer resp.Body.Close()
 
 	// Read response
 	body, err := io.ReadAll(resp.Body)
 	if err != nil {
-		log.Fatalf("Failed to read response: %v", err)
+		return "", fmt.Errorf("failed to read response: %v", err)
 	}
 
-	result, err := response.Parse(body, start)
+	// Check for error response
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("API error: %s - %s", resp.Status, string(body))
+	}
+
+	// Pretty print the JSON response
+	var prettyJSON bytes.Buffer
+	err = json.Indent(&prettyJSON, body, "", "  ")
 	if err != nil {
-		log.Fatal(err)
+		// If we can't pretty print, return the raw response
+		return string(body), nil
 	}
 
-	fmt.Printf("🤖 Model: %s\n", result.Model)
-	fmt.Printf("⏱️  Time: %.2fs\n", result.TimeTaken.Seconds())
-	fmt.Printf("💬 Response: %s\n", result.Response)
+	return prettyJSON.String(), nil
+}
+
+// createSignature creates an AWS Signature V4 for the request
+func createSignature(req *http.Request, config BedrockConfig, dateStamp, amzDate, payloadHash string) string {
+	// Step 1: Create a canonical request
+	canonicalURI := req.URL.Path
+	canonicalQueryString := req.URL.RawQuery
+
+	// Create canonical headers
+	canonicalHeaders := fmt.Sprintf("content-type:%s\nhost:%s\nx-amz-content-sha256:%s\nx-amz-date:%s\n",
+		req.Header.Get("Content-Type"),
+		req.Host,
+		payloadHash,
+		amzDate)
+
+	signedHeaders := "content-type;host;x-amz-content-sha256;x-amz-date"
+
+	// Create the canonical request
+	canonicalRequest := fmt.Sprintf("%s\n%s\n%s\n%s\n%s\n%s",
+		req.Method,
+		canonicalURI,
+		canonicalQueryString,
+		canonicalHeaders,
+		signedHeaders,
+		payloadHash)
+
+	// Debug output
+	fmt.Println("\n🔐 Canonical Request:")
+	fmt.Println(canonicalRequest)
+
+	// Step 2: Create a string to sign
+	credentialScope := fmt.Sprintf("%s/%s/%s/%s",
+		dateStamp,
+		config.Region,
+		awsService,
+		awsRequestType)
+
+	// Hash the canonical request
+	hashedCanonicalRequest := hashSHA256([]byte(canonicalRequest))
+
+	stringToSign := fmt.Sprintf("%s\n%s\n%s\n%s",
+		awsAlgorithm,
+		amzDate,
+		credentialScope,
+		hashedCanonicalRequest)
+
+	// Debug output
+	fmt.Println("\n🔍 String to Sign:")
+	fmt.Println(stringToSign)
+
+	// Step 3: Calculate the signature
+	// Create the signing key
+	kDate := hmacSHA256([]byte("AWS4"+config.SecretAccessKey), dateStamp)
+	kRegion := hmacSHA256(kDate, config.Region)
+	kService := hmacSHA256(kRegion, awsService)
+	kSigning := hmacSHA256(kService, awsRequestType)
```

Comment:
With mass assignments, it's nice to use a var block so they all line up. Like:

```suggestion
	var (
		kDate    = hmacSHA256([]byte("AWS4"+config.SecretAccessKey), dateStamp)
		kRegion  = hmacSHA256(kDate, config.Region)
		kService = hmacSHA256(kRegion, awsService)
		kSigning = hmacSHA256(kService, awsRequestType)
	)
```

or with shorter var names:

```go
	var (
		toSign = strings.Join([]string{awsAlgo, amzDate, credScope, canonReqHash}, "\")

		kDate    = hmacSHA256([]byte("AWS4"+config.SecretAccessKey), dateStamp)
		kRegion  = hmacSHA256(kDate, config.Region)
		kService = hmacSHA256(kRegion, awsService)
		kSigning = hmacSHA256(kService, awsRequestType)
	)
```

ps what is the k for, constants? that's not common in Go but not a prob.